app/src/main/java/es/craftsmanship/toledo/katangapp/activities/ShowStopsActivity.java

```diff
@@ -27,18 +26,15 @@
  */
 public class ShowStopsActivity extends BaseGeoLocatedActivity {
 
-    private List<BusStopResult> busStopResults;
     private RecyclerView recyclerView;
     private SwipeRefreshLayout swipeRefreshLayout;
 
     @Subscribe
     public void busStopsReceived(QueryResult queryResult) {
         List<BusStopResult> results = queryResult.getResults();
 
-        busStopResults.addAll(results);
-
-        recyclerView.setLayoutManager(new LinearLayoutManager(this));
```

Comment:
No, we aren't resetting the state of the recyclerView, just of the adapter. The recycler view is configured at the start to be a vertical list (linearlayoutmanager). We only need to change it or set it again if we want to show an horizontal list or a grid.